src/App.tsx

```diff
@@ -1,24 +1,391 @@
+                    }
+                    >
+                      <input
+                        type="text"
+                        className="todo__title-field"
+                        placeholder="Will be deleted if empty"
+                        onChange={handleTodoUpdated}
+                        value={newTitle}
+                      />
+                    </form>
+                  ) : (
+                    <>
+                      <span
+                        className="todo__title"
+                        onDoubleClick={() => handleDoubleClick(todo)}
+                      >
+                        {title}
+                      </span>
+
+                      <button
+                        type="button"
+                        className="todo__remove"
+                        onClick={() => deleteTodo(id)}
+                      >
+                        ×
+                      </button>
+                    </>
+                  )}
+
+                  <div className={classNames(
+                    'modal',
+                    'overlay',
+                    {
+                      'is-active':
+                        deleteTodoIds.includes(id)
+                        || updatedTodoIds.includes(id),
+                    },
+                  )}
+                  >
+                    <div className="
+                      modal-background
+                      has-background-white-ter
+                      "
+                    />
+                    <div className="loader" />
+                  </div>
+                </div>
+              );
+            })}
+
+            {query && isTodoLoaded && (
+              <div className="todo">
+                <label className="todo__status-label">
+                  <input type="checkbox" className="todo__status" />
+                </label>
+
+                <span className="todo__title">
+                  Todo is being saved now
+                </span>
+
+                <div className="modal overlay is-active">
+                  <div className="
+                    modal-background
+                    has-background-white-ter
+                    "
+                  />
+                  <div className="loader" />
+                </div>
+              </div>
+            )}
+          </section>
+        )}
+
+        {todos.length > 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              {`${activeTodos.length} ${activeTodos.length > 1 ? 'items' : 'item'} left`}
+            </span>
+
+            <nav className="filter">
+              <a
+                role="button"
+                href="#/"
+                className={classNames(
+                  'filter__link',
+                  'filter__link__all',
+                  { selected: filterType === Filter.ALL },
+                )}
+                onClick={() => handleFilterChange(Filter.ALL)}
```

Comment:
make one handler factory for all sort fields and just call in jsx wity proper value
```
const makeHandleFilterChange = (sortField: Filter) => () => handleFilterChange(sortField);
```
```suggestion
                onClick={(makeHandleFilterChange(Filter.ALL)}
```